development-setup-wizard.py

```diff
@@ -0,0 +1,244 @@
+            'Do you want replace it with a newly generated one? [y/n] ']))
+        if answer.lower() != 'y':
+            return
+
+    secret_key = get_secret_key()
+    contents = re.sub(r'SECRET_KEY=.*', f'SECRET_KEY={secret_key}', contents)
+    with open(filename, 'w') as f:
+        f.write(contents)
+    hooray(f'Successfully wrote secret key to {filename}')
+
+    # Rebuild if we added secret key
+    docker_compose_up(args)
+
+
+def get_secret_key():
+    cmd = ' '.join([
+        'docker compose exec -T django python -c',
+        '"from django.core.management.utils import get_random_secret_key;',
+        'print(get_random_secret_key())"'])
+    result = subprocess.run(cmd, shell=True, capture_output=True)
+    secret_key = (
+        result.stdout.decode().strip()
+        .replace("'", "\\'")
+        .replace("$", "_"))  # django-environ uses $ for interpolation
+
+    if not secret_key:
+        die(
+            "Could not successfully get secret key",
+            f"tried '{cmd}'"
+            f'\n{result.stderr.decode()}')
+    else:
+        hooray('Successfully got secret key')
+
+    return secret_key
+
+
+def create_django_superuser(args):
+    print(bold('Create your Django superuser:'))
+    subprocess.run([
+        'docker', 'compose', 'exec', 'django',
+        'python', 'manage.py', 'createsuperuser', '--noinput'])
+
+
+def load_sample_data(args):
+    print(bold('Loading Lark Camp sample data...'))
+    print('You will need to use your superuser credentials')
+    print('to complete this task')
+
+    subprocess.run([
+        'docker', 'compose', 'run', '--build', '--rm', 'data',
+        '/bin/sh', '-c', 'import'])
+
+    hooray('Successfully loaded Lark Camp sample data')
+
+
+def run_tests(args):
+    print(bold('Running backend tests'))
+
+    result = subprocess.run([
+        'docker', 'compose', 'exec', 'django', 'python', 'manage.py', 'test'])
+    if result.returncode != 0:
+        print(bold(red('Backend tests failed!')))
+        print(bold(' '.join([
+            'Although the process of setting up your development environment',
+            'succeeded, please try to find out why these test failures',
+            'happened and run the tests again.',
+        ])))
+    else:
+        hooray('Backend tests pass')
+
+
+def finish(args):
+    hooray('Development environment setup finished!')
+
+    print('\n'.join([
+        'For more information, see the following resources:',
+        '  https://github.com/camphoric/camphoric/blob/main/doc/development.md#frontend-development-process',
+        '  https://github.com/camphoric/camphoric/blob/main/doc/development.md#server-development-process',
+        '',
+        'To see your glorious work in action go to:',
+        '  http://localhost:3000',
+    ]))
+
+
+def die(title, *messages):
+    print(bold(red(title)))
+    for message in messages:
+        print(bold(message))
+    sys.exit(1)
+
+
+def hooray(title):
+    print(bold(green(f'✅ {title}')))
```

Comment:
I would avoid unusual characters in source code.  Instead output some stars (***) or color the text

Reply:
small amount of pushback - I'm not sure how "unsafe" this might be, I'm more inclined to say that if your terminal/editor/etc doesn't fully support utf8 that's more of a problem with the rendering software (and it's ability to be used in most non-english speaking countries) than with our usage of utf8 characters in code - but I'm interested in the counter-argument and don't feel too strongly about it.

Reply:
This will be the first script a new developer runs, and the process should be as smooth as possible and an encouraging introduction to the project. Rewriting in Python furthers that goal, and so does decreasing the required font support.
I agree that these days most people will have full utf-8 support -- even in the US :smiley:   But it is not universal. There will be some who cannot display non-ascii.  There will be more who can display European language accents, but not dingbats such as this checkmark. The further you stray from latin letters, the more likely you are to encounter incomplete support. I prefer a measured approach -- consider how much benefit you get from the utf-8 usage versus how much possible portability loss you incur. In this case, I think the advantage of displaying a check-mark is small.